### PR TITLE
[Finishes #104747280] Asset upload UI test for App Lab

### PR DESCRIPTION
**Pull request by joshlory**
View the Sauce Labs session here: https://saucelabs.com/tests/30b7b4e0c37746fe958f7d0a6350283b

### CODE REVIEW

File: dashboard/test/ui/features/applab/scenarios.feature [hunk 1]
```diff
@@ -16,7 +16,7 @@ Feature: App Lab Scenarios
   @no_mobile
   Scenario: App Lab Share
     # Create an app with a "hello world" button (omitting closing paren for auto-correct).
-    Then I press "show-code-header"
```
Comment (laurelfan) [inline dashboard/test/ui/features/applab/scenarios.feature:19]: is there any behavior difference for Then and When or is this just syntactic sugar to make it read better?

File: dashboard/test/ui/step_definitions/steps.rb [hunk 2]
```diff
@@ -461,3 +463,10 @@ def make_symbol_if_colon(key)
 Then /^there is no horizontal scrollbar$/ do
   @browser.execute_script('return document.documentElement.scrollWidth <= document.documentElement.clientWidth').should eq true
 end
+
+# Place files in dashboard/test/fixtures
+Then /^I upload the file named "(.*?)"$/ do |filename|
+  filename = File.expand_path(filename, '../fixtures')
```
Comment (laurelfan) [inline dashboard/test/ui/step_definitions/steps.rb:470]: whoa, it's this easy? nice

**Comment by laurelfan** [comment]
LGTM!
Have you tried it in all browsers yet?

You can run a feature on all browsers with `./runner.rb -f <feature_file> -p 10`
(do specify the feature, don't specify the browser, optionally specify ||-ness)

**Comment by laurelfan** [comment]
oh, and I noticed that it doesn't look like the image has loaded yet in the saucelabs log. Is there a way to check that a valid image loaded or that it is == to the one that was uploaded?

**Comment by joshlory** [comment]
Is an eyes test the best way to verify the image that gets uploaded?  The image does load, but the screenshot happened before the image finished loading.

File: dashboard/test/ui/features/applab/scenarios.feature (hunk 1 above)
Comment (joshlory) [inline dashboard/test/ui/features/applab/scenarios.feature:19]: Yeah they're the same.

**Comment by joshlory** [comment]
Scenario passing against all applicable browsers on Sauce Labs:

```
./runner.rb -d localhost-studio.code.org:3000 -p localhost.code.org:3000 --html -f features/applab/scenarios.feature:63 --parallel 10
UI tests for IE9Win7_applab_scenarios:63 skipped (1:01 minutes)
UI tests for iPhone_applab_scenarios:63 skipped (1:01 minutes)
UI tests for IE10Win7_applab_scenarios:63 skipped (1:02 minutes)
UI tests for iPad_applab_scenarios:63 skipped (1:02 minutes)
UI tests for SafariYosemite_applab_scenarios:63 skipped (1:02 minutes)
UI tests for Firefox39Win7_applab_scenarios:63 succeeded (2:01 minutes, 1 passed)
UI tests for Firefox38Yosemite_applab_scenarios:63 succeeded (2:19 minutes, 1 passed)
UI tests for ChromeLatestWin7_applab_scenarios:63 succeeded (2:31 minutes, 1 passed)
UI tests for Chrome44XP_applab_scenarios:63 succeeded (2:31 minutes, 1 passed)
```

**Comment by laurelfan** [comment]
LGTM!

**Pull request merged**